src/include/denc.h

```diff
@@ -242,10 +242,52 @@ WRITE_INT_DENC(bool, uint8_t);
 // high bit of each byte indicates another byte follows.
 template<typename T>
 inline void denc_varint(T v, size_t& p) {
-  p += sizeof(T) + 1;
+  p += sizeof(v) + 1;
 }
 
 template<typename T>
+inline void denc_varint_exact(T v, size_t& p) {
+  static_assert(std::is_unsigned<T>::value,
+                "denc_varint() operates on unsigned integers only");
+  register size_t i;
```

Comment:
I really doubt whether any change is probable here. Definitely FIXME.

Also, thanks for sharing the knowledge, @byo! I wasn't aware that `register` is going to be removed in C++17.